docs/slicing.md

```diff
@@ -0,0 +1,143 @@
+# Slicing
+
+Most sets in arrai can be sliced with the slice expression.
+
+The slicing expression can be used by putting the expression as an argument to a call.
+
+The syntax of the slicing expression is as follows:
+
+```text
+start=expr? ";" end=expr? (";" step=expr)?
+```
+
+As shown above, you can define a slice by defining a lower bound, an upper bound, both or none.
```

Comment:
Maybe some details about how the lower bound and upper bound are both inclusive?

Reply:
I'll put it in the behaviour section